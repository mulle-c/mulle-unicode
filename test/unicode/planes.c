#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>



static void  test( char *name,
                   int (*member_f)( mulle_utf32_t),
                   int (*plane_f)( unsigned int))
{
   mulle_utf32_t   c;
   unsigned int    planes[ 0x11];
   unsigned int    i;
   unsigned int    expect;

   memset( planes, 0, sizeof( planes));

   for( c = 0; c < 0x110000; c++)
      if( (*member_f)( c))
      {
         planes[ c >> 16] = 1;
         c  = c & ~0xFFFF;
         c += 0xFFFF;
         continue;
      }


   for( i = 0; i < 0x11; i++)
   {
      printf( "%s plane #%d : ", name, i);
      expect = (*plane_f)( i);
      if( expect == planes[ i])
         printf( "pass\n");
      else
         printf( "failed (expect: %d, result: %d)\n", expect, planes[ i]);
   }
}


int  main()
{
   test( "alphanumeric", mulle_utf32_is_alphanumeric, mulle_utf_is_alphanumericplane);
   test( "capitalized", mulle_utf32_is_capitalized, mulle_utf_is_capitalizedplane);
   test( "control", mulle_utf32_is_control, mulle_utf_is_controlplane);
   test( "decimaldigit", mulle_utf32_is_decimaldigit, mulle_utf_is_decimaldigitplane);
   test( "decomposable", mulle_utf32_is_decomposable, mulle_utf_is_decomposableplane);
   test( "legalcharacter", mulle_utf32_is_legalcharacter, mulle_utf_is_legalcharacterplane);
   test( "letter", mulle_utf32_is_letter, mulle_utf_is_letterplane);
   test( "lowercase", mulle_utf32_is_lowercase, mulle_utf_is_lowercaseplane);
   test( "newline", mulle_utf32_is_newline, mulle_utf_is_newlineplane);
   test( "nonbase", mulle_utf32_is_nonbase, mulle_utf_is_nonbaseplane);
   test( "noncharacter", mulle_utf32_is_noncharacter, mulle_utf_is_noncharacterplane);
   test( "nonpercentescape", mulle_utf32_is_nonpercentescape, mulle_utf_is_nonpercentescapeplane);
   test( "privatecharacter", mulle_utf32_is_privatecharacter, mulle_utf_is_privatecharacterplane);
   test( "punctuation", mulle_utf32_is_punctuation, mulle_utf_is_punctuationplane);
   test( "symbol", mulle_utf32_is_symbol, mulle_utf_is_symbolplane);
   test( "uppercase", mulle_utf32_is_uppercase, mulle_utf_is_uppercaseplane);
   test( "validurlfragment", mulle_utf32_is_validurlfragment, mulle_utf_is_validurlfragmentplane);
   test( "validurlhost", mulle_utf32_is_validurlhost, mulle_utf_is_validurlhostplane);
   test( "validurlpassword", mulle_utf32_is_validurlpassword, mulle_utf_is_validurlpasswordplane);
   test( "validurlpath", mulle_utf32_is_validurlpath, mulle_utf_is_validurlpathplane);
   test( "validurlquery", mulle_utf32_is_validurlquery, mulle_utf_is_validurlqueryplane);
   test( "validurlscheme", mulle_utf32_is_validurlscheme, mulle_utf_is_validurlschemeplane);
   test( "validurluser", mulle_utf32_is_validurluser, mulle_utf_is_validurluserplane);
   test( "whitespace", mulle_utf32_is_whitespace, mulle_utf_is_whitespaceplane);
   test( "whitespaceornewline", mulle_utf32_is_whitespaceornewline, mulle_utf_is_whitespaceornewlineplane);
   return( 0);
}

