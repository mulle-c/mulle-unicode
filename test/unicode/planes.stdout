alphanumeric plane #0 : pass
alphanumeric plane #1 : pass
alphanumeric plane #2 : pass
alphanumeric plane #3 : pass
alphanumeric plane #4 : pass
alphanumeric plane #5 : pass
alphanumeric plane #6 : pass
alphanumeric plane #7 : pass
alphanumeric plane #8 : pass
alphanumeric plane #9 : pass
alphanumeric plane #10 : pass
alphanumeric plane #11 : pass
alphanumeric plane #12 : pass
alphanumeric plane #13 : pass
alphanumeric plane #14 : pass
alphanumeric plane #15 : pass
alphanumeric plane #16 : pass
capitalized plane #0 : pass
capitalized plane #1 : pass
capitalized plane #2 : pass
capitalized plane #3 : pass
capitalized plane #4 : pass
capitalized plane #5 : pass
capitalized plane #6 : pass
capitalized plane #7 : pass
capitalized plane #8 : pass
capitalized plane #9 : pass
capitalized plane #10 : pass
capitalized plane #11 : pass
capitalized plane #12 : pass
capitalized plane #13 : pass
capitalized plane #14 : pass
capitalized plane #15 : pass
capitalized plane #16 : pass
control plane #0 : pass
control plane #1 : pass
control plane #2 : pass
control plane #3 : pass
control plane #4 : pass
control plane #5 : pass
control plane #6 : pass
control plane #7 : pass
control plane #8 : pass
control plane #9 : pass
control plane #10 : pass
control plane #11 : pass
control plane #12 : pass
control plane #13 : pass
control plane #14 : pass
control plane #15 : pass
control plane #16 : pass
decimaldigit plane #0 : pass
decimaldigit plane #1 : pass
decimaldigit plane #2 : pass
decimaldigit plane #3 : pass
decimaldigit plane #4 : pass
decimaldigit plane #5 : pass
decimaldigit plane #6 : pass
decimaldigit plane #7 : pass
decimaldigit plane #8 : pass
decimaldigit plane #9 : pass
decimaldigit plane #10 : pass
decimaldigit plane #11 : pass
decimaldigit plane #12 : pass
decimaldigit plane #13 : pass
decimaldigit plane #14 : pass
decimaldigit plane #15 : pass
decimaldigit plane #16 : pass
decomposable plane #0 : pass
decomposable plane #1 : pass
decomposable plane #2 : pass
decomposable plane #3 : pass
decomposable plane #4 : pass
decomposable plane #5 : pass
decomposable plane #6 : pass
decomposable plane #7 : pass
decomposable plane #8 : pass
decomposable plane #9 : pass
decomposable plane #10 : pass
decomposable plane #11 : pass
decomposable plane #12 : pass
decomposable plane #13 : pass
decomposable plane #14 : pass
decomposable plane #15 : pass
decomposable plane #16 : pass
legalcharacter plane #0 : pass
legalcharacter plane #1 : pass
legalcharacter plane #2 : pass
legalcharacter plane #3 : pass
legalcharacter plane #4 : pass
legalcharacter plane #5 : pass
legalcharacter plane #6 : pass
legalcharacter plane #7 : pass
legalcharacter plane #8 : pass
legalcharacter plane #9 : pass
legalcharacter plane #10 : pass
legalcharacter plane #11 : pass
legalcharacter plane #12 : pass
legalcharacter plane #13 : pass
legalcharacter plane #14 : pass
legalcharacter plane #15 : pass
legalcharacter plane #16 : pass
letter plane #0 : pass
letter plane #1 : pass
letter plane #2 : pass
letter plane #3 : pass
letter plane #4 : pass
letter plane #5 : pass
letter plane #6 : pass
letter plane #7 : pass
letter plane #8 : pass
letter plane #9 : pass
letter plane #10 : pass
letter plane #11 : pass
letter plane #12 : pass
letter plane #13 : pass
letter plane #14 : pass
letter plane #15 : pass
letter plane #16 : pass
lowercase plane #0 : pass
lowercase plane #1 : pass
lowercase plane #2 : pass
lowercase plane #3 : pass
lowercase plane #4 : pass
lowercase plane #5 : pass
lowercase plane #6 : pass
lowercase plane #7 : pass
lowercase plane #8 : pass
lowercase plane #9 : pass
lowercase plane #10 : pass
lowercase plane #11 : pass
lowercase plane #12 : pass
lowercase plane #13 : pass
lowercase plane #14 : pass
lowercase plane #15 : pass
lowercase plane #16 : pass
newline plane #0 : pass
newline plane #1 : pass
newline plane #2 : pass
newline plane #3 : pass
newline plane #4 : pass
newline plane #5 : pass
newline plane #6 : pass
newline plane #7 : pass
newline plane #8 : pass
newline plane #9 : pass
newline plane #10 : pass
newline plane #11 : pass
newline plane #12 : pass
newline plane #13 : pass
newline plane #14 : pass
newline plane #15 : pass
newline plane #16 : pass
nonbase plane #0 : pass
nonbase plane #1 : pass
nonbase plane #2 : pass
nonbase plane #3 : pass
nonbase plane #4 : pass
nonbase plane #5 : pass
nonbase plane #6 : pass
nonbase plane #7 : pass
nonbase plane #8 : pass
nonbase plane #9 : pass
nonbase plane #10 : pass
nonbase plane #11 : pass
nonbase plane #12 : pass
nonbase plane #13 : pass
nonbase plane #14 : pass
nonbase plane #15 : pass
nonbase plane #16 : pass
noncharacter plane #0 : pass
noncharacter plane #1 : pass
noncharacter plane #2 : pass
noncharacter plane #3 : pass
noncharacter plane #4 : pass
noncharacter plane #5 : pass
noncharacter plane #6 : pass
noncharacter plane #7 : pass
noncharacter plane #8 : pass
noncharacter plane #9 : pass
noncharacter plane #10 : pass
noncharacter plane #11 : pass
noncharacter plane #12 : pass
noncharacter plane #13 : pass
noncharacter plane #14 : pass
noncharacter plane #15 : pass
noncharacter plane #16 : pass
nonpercentescape plane #0 : pass
nonpercentescape plane #1 : pass
nonpercentescape plane #2 : pass
nonpercentescape plane #3 : pass
nonpercentescape plane #4 : pass
nonpercentescape plane #5 : pass
nonpercentescape plane #6 : pass
nonpercentescape plane #7 : pass
nonpercentescape plane #8 : pass
nonpercentescape plane #9 : pass
nonpercentescape plane #10 : pass
nonpercentescape plane #11 : pass
nonpercentescape plane #12 : pass
nonpercentescape plane #13 : pass
nonpercentescape plane #14 : pass
nonpercentescape plane #15 : pass
nonpercentescape plane #16 : pass
privatecharacter plane #0 : pass
privatecharacter plane #1 : pass
privatecharacter plane #2 : pass
privatecharacter plane #3 : pass
privatecharacter plane #4 : pass
privatecharacter plane #5 : pass
privatecharacter plane #6 : pass
privatecharacter plane #7 : pass
privatecharacter plane #8 : pass
privatecharacter plane #9 : pass
privatecharacter plane #10 : pass
privatecharacter plane #11 : pass
privatecharacter plane #12 : pass
privatecharacter plane #13 : pass
privatecharacter plane #14 : pass
privatecharacter plane #15 : pass
privatecharacter plane #16 : pass
punctuation plane #0 : pass
punctuation plane #1 : pass
punctuation plane #2 : pass
punctuation plane #3 : pass
punctuation plane #4 : pass
punctuation plane #5 : pass
punctuation plane #6 : pass
punctuation plane #7 : pass
punctuation plane #8 : pass
punctuation plane #9 : pass
punctuation plane #10 : pass
punctuation plane #11 : pass
punctuation plane #12 : pass
punctuation plane #13 : pass
punctuation plane #14 : pass
punctuation plane #15 : pass
punctuation plane #16 : pass
symbol plane #0 : pass
symbol plane #1 : pass
symbol plane #2 : pass
symbol plane #3 : pass
symbol plane #4 : pass
symbol plane #5 : pass
symbol plane #6 : pass
symbol plane #7 : pass
symbol plane #8 : pass
symbol plane #9 : pass
symbol plane #10 : pass
symbol plane #11 : pass
symbol plane #12 : pass
symbol plane #13 : pass
symbol plane #14 : pass
symbol plane #15 : pass
symbol plane #16 : pass
uppercase plane #0 : pass
uppercase plane #1 : pass
uppercase plane #2 : pass
uppercase plane #3 : pass
uppercase plane #4 : pass
uppercase plane #5 : pass
uppercase plane #6 : pass
uppercase plane #7 : pass
uppercase plane #8 : pass
uppercase plane #9 : pass
uppercase plane #10 : pass
uppercase plane #11 : pass
uppercase plane #12 : pass
uppercase plane #13 : pass
uppercase plane #14 : pass
uppercase plane #15 : pass
uppercase plane #16 : pass
validurlfragment plane #0 : pass
validurlfragment plane #1 : pass
validurlfragment plane #2 : pass
validurlfragment plane #3 : pass
validurlfragment plane #4 : pass
validurlfragment plane #5 : pass
validurlfragment plane #6 : pass
validurlfragment plane #7 : pass
validurlfragment plane #8 : pass
validurlfragment plane #9 : pass
validurlfragment plane #10 : pass
validurlfragment plane #11 : pass
validurlfragment plane #12 : pass
validurlfragment plane #13 : pass
validurlfragment plane #14 : pass
validurlfragment plane #15 : pass
validurlfragment plane #16 : pass
validurlhost plane #0 : pass
validurlhost plane #1 : pass
validurlhost plane #2 : pass
validurlhost plane #3 : pass
validurlhost plane #4 : pass
validurlhost plane #5 : pass
validurlhost plane #6 : pass
validurlhost plane #7 : pass
validurlhost plane #8 : pass
validurlhost plane #9 : pass
validurlhost plane #10 : pass
validurlhost plane #11 : pass
validurlhost plane #12 : pass
validurlhost plane #13 : pass
validurlhost plane #14 : pass
validurlhost plane #15 : pass
validurlhost plane #16 : pass
validurlpassword plane #0 : pass
validurlpassword plane #1 : pass
validurlpassword plane #2 : pass
validurlpassword plane #3 : pass
validurlpassword plane #4 : pass
validurlpassword plane #5 : pass
validurlpassword plane #6 : pass
validurlpassword plane #7 : pass
validurlpassword plane #8 : pass
validurlpassword plane #9 : pass
validurlpassword plane #10 : pass
validurlpassword plane #11 : pass
validurlpassword plane #12 : pass
validurlpassword plane #13 : pass
validurlpassword plane #14 : pass
validurlpassword plane #15 : pass
validurlpassword plane #16 : pass
validurlpath plane #0 : pass
validurlpath plane #1 : pass
validurlpath plane #2 : pass
validurlpath plane #3 : pass
validurlpath plane #4 : pass
validurlpath plane #5 : pass
validurlpath plane #6 : pass
validurlpath plane #7 : pass
validurlpath plane #8 : pass
validurlpath plane #9 : pass
validurlpath plane #10 : pass
validurlpath plane #11 : pass
validurlpath plane #12 : pass
validurlpath plane #13 : pass
validurlpath plane #14 : pass
validurlpath plane #15 : pass
validurlpath plane #16 : pass
validurlquery plane #0 : pass
validurlquery plane #1 : pass
validurlquery plane #2 : pass
validurlquery plane #3 : pass
validurlquery plane #4 : pass
validurlquery plane #5 : pass
validurlquery plane #6 : pass
validurlquery plane #7 : pass
validurlquery plane #8 : pass
validurlquery plane #9 : pass
validurlquery plane #10 : pass
validurlquery plane #11 : pass
validurlquery plane #12 : pass
validurlquery plane #13 : pass
validurlquery plane #14 : pass
validurlquery plane #15 : pass
validurlquery plane #16 : pass
validurlscheme plane #0 : pass
validurlscheme plane #1 : pass
validurlscheme plane #2 : pass
validurlscheme plane #3 : pass
validurlscheme plane #4 : pass
validurlscheme plane #5 : pass
validurlscheme plane #6 : pass
validurlscheme plane #7 : pass
validurlscheme plane #8 : pass
validurlscheme plane #9 : pass
validurlscheme plane #10 : pass
validurlscheme plane #11 : pass
validurlscheme plane #12 : pass
validurlscheme plane #13 : pass
validurlscheme plane #14 : pass
validurlscheme plane #15 : pass
validurlscheme plane #16 : pass
validurluser plane #0 : pass
validurluser plane #1 : pass
validurluser plane #2 : pass
validurluser plane #3 : pass
validurluser plane #4 : pass
validurluser plane #5 : pass
validurluser plane #6 : pass
validurluser plane #7 : pass
validurluser plane #8 : pass
validurluser plane #9 : pass
validurluser plane #10 : pass
validurluser plane #11 : pass
validurluser plane #12 : pass
validurluser plane #13 : pass
validurluser plane #14 : pass
validurluser plane #15 : pass
validurluser plane #16 : pass
whitespace plane #0 : pass
whitespace plane #1 : pass
whitespace plane #2 : pass
whitespace plane #3 : pass
whitespace plane #4 : pass
whitespace plane #5 : pass
whitespace plane #6 : pass
whitespace plane #7 : pass
whitespace plane #8 : pass
whitespace plane #9 : pass
whitespace plane #10 : pass
whitespace plane #11 : pass
whitespace plane #12 : pass
whitespace plane #13 : pass
whitespace plane #14 : pass
whitespace plane #15 : pass
whitespace plane #16 : pass
whitespaceornewline plane #0 : pass
whitespaceornewline plane #1 : pass
whitespaceornewline plane #2 : pass
whitespaceornewline plane #3 : pass
whitespaceornewline plane #4 : pass
whitespaceornewline plane #5 : pass
whitespaceornewline plane #6 : pass
whitespaceornewline plane #7 : pass
whitespaceornewline plane #8 : pass
whitespaceornewline plane #9 : pass
whitespaceornewline plane #10 : pass
whitespaceornewline plane #11 : pass
whitespaceornewline plane #12 : pass
whitespaceornewline plane #13 : pass
whitespaceornewline plane #14 : pass
whitespaceornewline plane #15 : pass
whitespaceornewline plane #16 : pass
